hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/volumes/volumes.tsx

```diff
@@ -0,0 +1,422 @@
+      console.log('Not a valid option')
+    }
+  }
+
+  _getSelectedColumns = (selected: IOption[]) => {
+    const selectedColumns = selected.length > 0 ? selected : COLUMNS.filter(column => column.isVisible).map(column => ({
+      label: column.key,
+      value: column.key
+    }));
+    return selectedColumns;
+  };
+
+  _handleAclLinkClick = (volume: IVolume) => {
+    this.setState({
+      showPanel: true,
+      currentRow: volume
+    });
+  };
+
+  _loadData = () => {
+    this.setState(prevState => ({
+      loading: true,
+      selectedColumns: this._getSelectedColumns(prevState.selectedColumns),
+      showPanel: false
+    }));
+    const { request, controller } = AxiosGetHelper('/api/v1/volumes', cancelSignal,
+        "", { limit: this.state.selectedLimit.value});
+    cancelSignal = controller;
+    request.then(response => {
+      const volumesResponse: IVolumesResponse = response.data;
+      const totalCount = volumesResponse.totalCount;
+      const volumes: IVolumeResponse[] = volumesResponse.volumes;
+      const dataSource: IVolume[] = volumes.map(volume => {
+        return {
+          volume: volume.volume,
+          owner: volume.owner,
+          admin: volume.admin,
+          creationTime: volume.creationTime,
+          modificationTime: volume.modificationTime,
+          quotaInBytes: volume.quotaInBytes,
+          quotaInNamespace: volume.quotaInNamespace,
+          usedNamespace: volume.usedNamespace,
+          acls: volume.acls
+        };
+      });
+
+      this.setState({
+        loading: false,
+        dataSource,
+        totalCount,
+        lastUpdated: Number(moment()),
+        showPanel: false
+      });
+    }).catch(error => {
+      this.setState({
+        loading: false,
+        showPanel: false
+      });
+      showDataFetchError(error.toString());
+    });
+  };
+
+  componentDidMount(): void {
+    // Fetch volumes on component mount
+    this._loadData();
+    this.autoReload.startPolling();
+  }
+
+  componentWillUnmount(): void {
+    this.autoReload.stopPolling();
+    cancelSignal && cancelSignal.abort();
+  }
+
+  onShowSizeChange = (current: number, pageSize: number) => {
+    console.log(current, pageSize);
+  };
+
+  render() {
+    const {dataSource, loading, totalCount, lastUpdated, selectedColumns,
+      columnOptions, showPanel, currentRow, selectedLimit} = this.state;
+    const paginationConfig: PaginationConfig = {
+      showTotal: (total: number, range) => `${range[0]}-${range[1]} of ${total} volumes`,
+      showSizeChanger: true,
+      onShowSizeChange: this.onShowSizeChange
+    };
+    return (
+      <div className='volumes-container'>
+        <div className='page-header'>
+          Volumes ({totalCount})
+          <div className='filter-block'>
+            <MultiSelect
+              allowSelectAll
+              isMulti
+              maxShowValues={3}
+              className='multi-select-container'
+              options={columnOptions}
```

Comment:
So `defaultColumns` also includes Volume column.
This causes the volume column to be de-selectable as well. However since the page is showing Volume information, we should exclude that from the selection - as in the Volume column should not be de-selectable from the view.
<img width="300" alt="Screenshot 2023-09-26 at 11 56 10 AM" src="https://github.com/apache/ozone/assets/43001336/38d1daf2-562a-4d5a-a6ef-6ffeb998aaaa">

Reply:
Thanks for the review. I have added `isOptionDisabled` for volume, so that volume cannot be deselected and always chosen.

Reply:
<img width="182" alt="image" src="https://github.com/apache/ozone/assets/36403683/0cf0970f-6e25-4864-944f-eacd22c66cc3">
For reference